pybamm/expression_tree/unary_operators.py

```diff
@@ -164,22 +164,35 @@ class NumpyBroadcast(Broadcast):
 
     def __init__(self, child, domain, mesh):
         super().__init__(child, domain, name="numpy broadcast")
+        # determine broadcasting vector size (size 1 if the domain is empty)
+        if domain == []:
+            broadcasting_vector_size = 1
+        else:
+            broadcasting_vector_size = sum([mesh[dom].npts for dom in domain])
         # create broadcasting vector (vector of ones with shape determined by the
         # domain)
-        broadcasting_vector_size = sum([mesh[dom].npts for dom in domain])
         self.broadcasting_vector = np.ones(broadcasting_vector_size)
 
     def evaluate(self, t=None, y=None):
         """ See :meth:`pybamm.Symbol.evaluate()`. """
+        child = self.children[0]
+        child_eval = child.evaluate(t, y)
         # if child is a vector, add a dimension for broadcasting
-        if isinstance(self.children[0], pybamm.Vector):
-            return (
-                self.children[0].evaluate(t, y)[:, np.newaxis]
-                * self.broadcasting_vector
+        if isinstance(child, pybamm.Vector):
+            return child_eval[:, np.newaxis] * self.broadcasting_vector
+        # if child is a state vector, check that it has the right shape and then
+        # transpose, broadcast, transpose again
+        # QUESTION: is there a better way of doing this than transposing twice?
+        elif isinstance(child, pybamm.StateVector):
+            assert child_eval.shape[0] == 1, ValueError(
+                """child_eval should have shape (1,n), not {}""".format(
+                    child_eval.shape
+                )
             )
+            return (child_eval.T * self.broadcasting_vector).T
```

Comment:
could use `np.repeat(child_eval,broadcasting_vector_size,axis=0)`?

Reply:
Cool, this is a bit faster and clearer so have updated. This `evaluate` still feels a bit hacky but will just improve as and when needed